### PR TITLE
XTRA Unity MetaData Extension

### PR DESCRIPTION
**Pull request by MackeyK24**

### CODE REVIEW

**Comment by donmccurdy** [comment]
I'd recommend reserving the prefix as well — is "XTRA" a brand or application name, or a reference to the "extra metadata" purpose of this extension? See https://github.com/KhronosGroup/glTF/blob/master/extensions/Prefixes.md. Is Babylon Toolkit associated with BabylonJS? I might suggest "BABYLON" if so, and "BABYLONTOOLKIT" if not.

**Comment by MackeyK24** [comment]
Well Babylon is just one implementation... I also have PlayCanvas and might throw ThreeJS as well

I like XTRA .... can I not use that anyways ???

**Comment by MackeyK24** [comment]
Ok... I changed my prefix to CVTOOLS

**Comment by robertlong** [comment]
I'm also looking into storing entity component data in a glTF for use in Mozilla Hubs (see #1387).

I noticed this extension stores the data on the `extras` property without a namespace. Using this extension essentially means: treat `node.extras.metadata` as Unity metadata.

I think if this is going to be an extension then it should store that metadata in an extension object on the node. However, I am still confused by what `extras` is to be used for if not for this type of application specific metadata.

Perhaps we can work together on figuring out a best practice for storing this type of data? I'm excited to see other people looking into this!

**Comment by MackeyK24** [comment]
From GLTF readme

Extensions vs. Extras

In addition to extensions, the extras object can also be used to extend glTF. This is completely separate from extensions.

All glTF object properties allow adding new properties to an extras object sub-property, e.g.,

{ "shaders": { "a-vertex-shader-id": { "name": "user-defined-name", "uri" : "vertex-shader.glsl", "type": 35633, "extras" : { "precompiled_binary" : "path_to_precompiled_shader" } } } }
This enables glTF models to contain application-specific properties without creating a full glTF extension. This may be preferred for niche use cases where an extension would not be widely adopted.

**Comment by MackeyK24** [comment]
Basically I want extension so I can ID the glTF has unity metadata ... but because of detail I encode as EXTRA metadata

**Comment by MackeyK24** [comment]
OK... The CVTOOLS prefix has been registered :)

**Comment by pjcozzi** [comment]
@MackeyK24 is this extension implemented in Canvas Tools or anywhere else?

@robertlong is this still of interest to your use case?

File: extensions/2.0/Vendor/CVTOOLS_unity_metadata/README.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    {
+      "mesh": 2,
+      "scale": [
+        1.25,
+        1.25,
+        1.25
+      ],
+      "name": "Sphere",
+      "extras": {
+        "metadata": {
```
Comment (aaronfranke) [inline extensions/2.0/Vendor/CVTOOLS_unity_metadata/README.md:51]: glTF extensions are not supposed to specify what goes in `"extras"`.

**Review by aaronfranke — COMMENTED** [review]
Given that this extension is abandoned, the "Canvas Tools" project this was implemented in is abandoned, the extension is malformed in several ways, and the extension is inherently unstructured, I believe this extension should be rejected and the PR should be closed.

**Pull request closed without merging**